apps/condo/domains/marketplace/access/MarketItemFile.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Generated by `createschema marketplace.MarketItemFile 'marketItem:Relationship:MarketItem:CASCADE; file:File;'`
+ */
+
+const { get } = require('lodash')
+
+const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
+const { getById } = require('@open-condo/keystone/schema')
+
+const { checkPermissionInUserOrganizationOrRelatedOrganization } = require('@condo/domains/organization/utils/accessSchema')
+const { RESIDENT } = require('@condo/domains/user/constants/common')
+const { STAFF } = require('@condo/domains/user/constants/common')
+
+async function canManageMarketItemFiles ({ authentication: { item: user }, originalInput, operation, itemId }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin || user.isSupport) return true
+
+    if (user.type === RESIDENT) return false
+
+    if (user.type === STAFF) {
+        if (operation === 'create') {
+            const marketItemId = get(originalInput, ['marketItem', 'connect', 'id'], null)
+
+            if (marketItemId) {
+                const marketItem = await getById('MarketItem', marketItemId)
+                const organizationId = get(marketItem, 'organization', null)
+
+                if (!organizationId) return false
+
+                return await checkPermissionInUserOrganizationOrRelatedOrganization(user.id, organizationId, 'canManageMarketItems')
+            }
+
+            return false
+        }
+
+        const marketItemFile = await getById('MarketItemFile', itemId)
+        if (!marketItemFile) return false
+
+        const { organization } = marketItemFile
+        if (!organization) return false
+
+        return await checkPermissionInUserOrganizationOrRelatedOrganization(user.id, organization, 'canManageMarketItems')
```

Comment:
Please refactor logic to call checking of permissions only once. Maybe something like the following:
```suggestion
        let marketItem
        if (operation === 'create') {
            const marketItemId = get(originalInput, ['marketItem', 'connect', 'id'], null)
            if (marketItemId) {
               marketItem = await getById('MarketItem', marketItemId)
            }
        } else if (operation === 'update') {
            const marketItemFile = await getById('MarketItemFile', itemId)
            if (marketItemFile) {
                marketItem = await getById('MarketItem', marketItemFile.marketItem)
            }
        }
        
        organizationId = get(marketItem, 'organization', null)
        if (!organizationId) return false

        return await checkPermissionInUserOrganizationOrRelatedOrganization(user.id, organizationId, 'canManageMarketItems')
```